### ISSUE TITLE
[Bug]: Failed to load TTS engines

### ISSUE THREAD

**Issue by delargewiu** [ImperialSquid/zotero-zotts#98]
### Checklist

- [X] I have used the search function for [**OPEN**](https://github.com/ImperialSquid/zotero-zotts/issues) issues to see if someone else has already submitted the same bug report.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/ImperialSquid/zotero-zotts/issues?q=is%3Aissue+is%3Aclosed) issues to see if the problem is already solved and just waiting to be released.
- [X] I have described the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain paper, I will include its details.

### Zotero version

7.0.3

### ZoTTS version

1.2.1

### OS

Linux

### OS (specific)

Linux Mint 22 Cinnamon 6.2.9, Linux Kernel 6.8.0-40-generic

### Steps to reproduce

1. Installed the ZoTTS plugin
2. Close and open Zotero

### Expected behaviour

Open with ZoTTS GUI buttons available

### Actual behaviour

Zotero opens with a dialog box that states, "ZoTTS, ERROR: ZoTTS failed to load TTS engines. Please submit a bug report for assistance." The GUI buttons for ZoTTS are also missing. This issue is identical on two different computers running the above-stated operating system.

**Comment by ImperialSquid** [comment]
Hi @delargewiu, thanks for your bug report! 

Can you try going to [this website](https://mdn.github.io/dom-examples/web-speech-api/speak-easy-synthesis/) and confirming if you see any voices listed in the dropdown?

**Comment by ImperialSquid** [comment]
Hi @delargewiu, thanks for your bug report! 

Can you try going to [this website](https://mdn.github.io/dom-examples/web-speech-api/speak-easy-synthesis/) and confirming if you see any voices listed in the dropdown?

If you don't see any voices it means you don't have a way to do TTS on your machines. You'll need to install [speech-dispatcher](https://community.linuxmint.com/software/view/speech-dispatcher) as well as some kind of engine like [espeak](https://community.linuxmint.com/software/view/espeak).

**Comment by HynDuf** [comment]
I got the same bug too (after trying to install it for the first time). And there are 6 voices listed in the dropdown normally for me.

**Comment by ImperialSquid** [comment]
@HynDuf due to the different distros I'll sort this in a new issue (it may be that the solutions end up being the same but just to not confuse things keeping them as separate seems easier)

**Comment by delargewiu** [comment]
@ImperialSquid 

I had Zotero installed using apt from a repository. I purged that and am now running from the extracted tarball as described on the Zotero website. Issue was still there.  I was able to go to the website and see the voices and use the tool.  Still had the issue. My system did have speech-dispatcher but not espeak so I installed it. I still have the issue.

**Comment by ImperialSquid** [comment]
Hmmm, annoying that it wasn't an easy fix but oh well lol

In Zotero, from the top menu bar, go to Tools > Developer > Error Console, then paste the following code in and run it:

```
var utt = new window.SpeechSynthesisUtterance("test")
utt.onstart = () => {console.log("start")}
utt.onend = () => {console.log("end")}
utt.onerror = (error) => {console.log(error.error)}
window.speechSynthesis.speak(utt)
```

Let me know what output you see, may need to give it a couple of seconds for something to show up

(Future versions of ZoTTS will have better error reporting)

**Comment by trhi** [comment]
Hi, I had the same problem (macOS).

Installed the plugin. It said installation failed. I did what you advised above, heard the voice saying 'test'.

I uninstalled the plugin, installed it again and it worked.

**Comment by Dav1com** [comment]
Hi, same issue here.
Tried the website and I have 1 voice listed, heard the voice saying "test". The error popup displays no cause.
I'm on ArchLinux with Zotero 7.0.5 (--version says 115.14.0esr).

**Comment by ImperialSquid** [comment]
@trhi so just to confirm, you're not having issues any more? 

@Dav1com since failed start ups can have a variety of causes, please open a new issue just to avoid confusion and cross talk while we diagnose everything (I know the bug template says about checking open issues, sorry for the confusion!)

**Comment by trhi** [comment]
That's right. In my case it seemed to help to just uninstall and install again.
